Add documentation related to service accounts (currently in beta) (#27)
1Password/kubernetes-secrets-injector: README.md

The 1Password Secrets Injector implements a mutating webhook to inject 1Password secrets as environment variables into a Kubernetes pod or deployment. Unlike the [1Password Kubernetes Operator](https://github.com/1Password/onepassword-operator), the Secrets Injector doesn't create a Kubernetes Secret when assigning secrets to your resource.

> The 1Password Secrets Injector for Kubernetes can use [1Password Connect](https://developer.1password.com/docs/connect) or [1Password Service Accounts](https://developer.1password.com/docs/service-accounts) to retrieve items.

Reviewer: Should we add this here as well? 

```suggestion
The 1Password Secrets Injector for Kubernetes can use [1Password Connect](https://developer.1password.com/docs/connect) or [1Password Service Accounts](https://developer.1password.com/docs/service-accounts)<sup>BETA</sup> to retrieve items.
```

            value: op://my-vault/my-item/sql/password
```

<details>
<summary>Usage with 1Password Service Accounts <sup>BETA</sup></summary>

```yaml

> # client-deployment.yaml - you client deployment/pod where you want to inject secrets

Reviewer: ```suggestion
# client-deployment.yaml - The client deployment/pod where you want to inject secrets
```

apiVersion: apps/v1
kind: Deployment
metadata:
  name: app-example
spec:
  selector:
    matchLabels:
      app: app-example
  template:
    metadata:
      annotations:
        operator.1password.io/inject: "app-example1"
        operator.1password.io/version: "2-beta"
      labels:
        app: app-example
    spec:
      containers:
        - name: app-example1
          image: my-image
          ports:
            - containerPort: 5000
          command: ["npm"]
          args: ["start"]

>           # This app will have the secrets injected using Service Accounts (currently in beta).

Reviewer: ```suggestion
          # A 1Password Service Account (currently in beta) will inject secrets into this application.
```

          env:
          - name: OP_SERVICE_ACCOUNT_TOKEN
            valueFrom:
              secretKeyRef:
                name: op-service-account
                key: token
          - name: DB_USERNAME
            value: op://my-vault/my-item/sql/username
          - name: DB_PASSWORD
            value: op://my-vault/my-item/sql/password

>         - name: my-app # because my-app is not listed in the inject annotation above this container will not be injected with secrets

Reviewer: ```suggestion
        - name: my-app # my-app isn't listed in the inject annotation above, so secrets won't be injected into this container.
```

## Setup and Deployment

### Prerequisites
- [docker installed](https://docs.docker.com/get-docker/)
- [kubectl installed](https://kubernetes.io/docs/tasks/tools/install-kubectl/)

> If you want to go with 1Password Connect, you'll need to:

Reviewer: ```suggestion
If you want to use 1Password Connect:
```

- [Set up a Secrets Automation workflow](https://developer.1password.com/docs/connect/get-started#step-1-set-up-a-secrets-automation-workflow).
- [Deploy 1Password Connect](https://developer.1password.com/docs/connect/get-started#step-2-deploy-1password-connect-server) in your Kubernetes infrastructure.

> Then, follow the [Use with 1Password Connect guide](#use-with-1password-connect).

Reviewer: ```suggestion
Then, follow instructions to [use the Kubernetes Injector with 1Password Connect](#use-with-1password-connect).
```

> If you want to go with 1Password Service Accounts <sup>BETA</sup>, you'll need to:

Reviewer: ```suggestion
If you want to use 1Password Service Accounts <sup>BETA</sup>:
```

- [Create a service account.](https://developer.1password.com//docs/service-accounts/)

> Then, follow the [Use with Service Account guide](#use-with-1password-service-accounts-supbetasup).

Reviewer: ```suggestion
Then, follow instructions to [use the Kubernetes Injector with a service account](#use-with-1password-service-accounts-supbetasup).
```

    value: op://my-vault/my-item/sql/username
```

## Use with 1Password Service Accounts <sup>BETA</sup>

### Step 1: Create a Kubernetes secret containing `OP_SERVICE_ACCOUNT_TOKEN`
```
kubectl create secret generic op-service-account --from-literal=token=YOUR_OP_SERVICE_ACCOUNT_TOKEN
```

### Step 2: Add the label `secrets-injection=enabled` label to the namespace:
```
kubectl label namespace default secrets-injection=enabled
```

### Step 3: Deploy injector
```
make deploy
```
**NOTE:** The injector creates the TLS certificate required for the webhook to work on the fly when deploying the injector (`deployment.yaml`). When the injector is removed from the cluster, it will delete the certificate.

### Step 4: Annotate your client pod or deployment with `inject` annotation

> Annotate your client pod/deployment spec with `operator.1password.io/inject` which expects a comma separated list of the names of the containers to that will be mutated and have secrets injected.

Reviewer: ```suggestion
Annotate your client pod or deployment spec with `operator.1password.io/inject`l. It expects a comma separated list of the names of the containers that will be mutated and have secrets injected.
```

```yaml
# client-deployment.yaml
annotations:
  operator.1password.io/inject: "app-example1"
```

### Step 5: Annotate your client pod or deployment with `version` annotation

> Annotate your client pod/deployment with the latest 1Password CLI beta version (`2.16.0-beta.01` or later).

Reviewer: ```suggestion
Annotate your client pod or deployment with the latest 1Password CLI beta version (`2.16.0-beta.01` or later).
```

```yaml
# client-deployment.yaml
annotations:
  operator.1password.io/version: "2-beta"
```

### Step 6: Configure the resource's environment

> Add an environment variable to the resource with a value referencing your 1Password item in the format `op://<vault>/<item>[/section]/<field>`.

Reviewer: ```suggestion
Add an environment variable to the resource with a value referencing your 1Password item. Use the following secret reference syntax: `op://<vault>/<item>[/section]/<field>`.
```
